app/scripts/metamask-controller.js

```diff
@@ -4796,6 +4796,33 @@ export default class MetamaskController extends EventEmitter {
     } ${hdPathDescription || ''}`.trim();
   }
 
+  /**
+   * Set account name that checks for identical label
+   *
+   * @param {string} accountId - Account ID
+   * @param {number} index - Index of the account in the accounts list
+   * @param {string} hardwareDeviceName - Name of the hardware device (Onekey, Ledger, etc)
+   * @param {string} hdPathDescription - HD path description
+   */
+  setAccountName(accountId, index, hardwareDeviceName, hdPathDescription) {
+    const label = this.getAccountLabel(
+      hardwareDeviceName,
+      index,
+      hdPathDescription,
+    );
+    try {
+      this.accountsController.setAccountName(accountId, label);
+    } catch {
+      const newIndex = index + 1;
+      this.setAccountName(
+        accountId,
+        newIndex,
+        hardwareDeviceName,
+        hdPathDescription,
+      );
+    }
+  }
```

Comment:
I don't think `try`/`catch` is the right tool here. It could throw any error and potentially send this code into an infinite loop (well, just until we blow the stack because of the recursion here).

Is there a `this.accountsController.getAccount` function we can call to check if the name is already taken, and then incremente it only in that case?